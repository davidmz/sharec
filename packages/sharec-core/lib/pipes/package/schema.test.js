const { fixtures } = require('testUtils')
const { packageJson } = require('./schema')

describe('pipes > package > schema', () => {
  describe('JSON', () => {
    const packageBaseFxt = fixtures('package/json/00-base', 'map')

    it('should merge configs', () => {
      expect(packageJson(packageBaseFxt)).toMatchSnapshot()
    })
  })
})
