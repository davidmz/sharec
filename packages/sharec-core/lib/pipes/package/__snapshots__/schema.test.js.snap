// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipes > package > schema JSON should merge configs 1`] = `
Map {
  "scripts" => Map {
    "api" => "documentation build *.js -f md -o api.md",
    "spell" => "yarn api && yaspeller-ci *.md",
    "clean" => "rimraf api.md coverage/",
    "test" => "jest --coverage && eslint-ci . && size-limit && yarn spell",
  },
  "devDependencies" => Map {
    "@logux/eslint-config" => "^33.0.0",
    "clean-publish" => "^1.1.2",
    "documentation" => "^12.1.2",
    "eslint" => "^6.5.1",
    "eslint-ci" => "^1.0.0",
    "eslint-plugin-es5" => "^1.4.1",
    "eslint-plugin-import" => "^2.18.2",
    "eslint-plugin-jest" => "^22.20.0",
    "eslint-plugin-node" => "^10.0.0",
    "eslint-plugin-promise" => "^4.2.1",
    "eslint-plugin-security" => "^1.4.0",
    "eslint-plugin-standard" => "^4.0.0",
    "eslint-config-standard" => "^13.0.1",
    "husky" => "^3.0.9",
    "jest" => "^24.9.0",
    "lint-staged" => "^9.4.2",
    "rimraf" => "^2.6.3",
    "size-limit" => "^1.3.7",
    "yaspeller-ci" => "^1.0.1",
    "@size-limit/preset-small-lib" => "^2.1.6",
    "eslint-plugin-prefer-let" => "^1.0.1",
    "eslint-plugin-unicorn" => "^12.1.0",
  },
  "husky" => Map {
    "hooks" => Map {
      "pre-commit" => "lint-staged",
    },
  },
  "lint-staged" => Map {
    "*.md" => "yaspeller-ci",
    "*.js" => "eslint",
  },
  "jest" => Map {
    "testEnvironment" => "node",
    "coverageThreshold" => Map {
      "global" => Map {
        "statements" => 100,
      },
    },
  },
  "eslintConfig" => Map {
    "extends" => "@logux/eslint-config/browser",
  },
  "eslintIgnore" => Array [
    "node_modules",
  ],
  "yaspeller" => Map {
    "lang" => "en",
    "ignoreCapitalization" => true,
    "ignoreText" => Array [
      " \\\\(by [^)]+\\\\).",
    ],
  },
}
`;
